modules/test/application/clicommands/PhpCommand.php

```diff
@@ -155,6 +158,96 @@ public function styleAction()
         );
     }
 
+    /**
+     * Run code-validity checks
+     *
+     * This command checks whether icingaweb and installed modules match PHP syntax.
+     *
+     * USAGE
+     *
+     * icingacli test php validity
+     */
+    public function validityAction()
+    {
+        $types = array(
+            T_CLASS     => 'class',
+            T_INTERFACE => 'interface'
+        );
+
+        if (version_compare(PHP_VERSION, '5.4.0') > -1) {
+            $types[T_TRAIT] = 'trait';
+        }
+
+        $files = array();
+        $baseDir = realpath(__DIR__ . '/../../../..');
```

Comment:
Is there any reason not to base this on Icinga\Application methods?

Reply:
Yes: This is PhpUnit, not our fully-bootstrapped CLI.